src/TuxxAnimations.js

```diff
@@ -49,59 +50,323 @@ var createAnimationClass = function (transitions, customClassName) {
     },
 
     componentWillMount: function () {
-      //Change all custom props or add them if prop not defined in default
+      //set enter and leave props
+      var enterProps = this.props.enter;
```

Comment:
This method has a lot of non-DRY looking code to me. It could use a refactor to DRY things up and some of the functionality in this should be broken out into separate functions that will be captured via closure scope.